packages/contracts-bedrock/scripts/DeployPeriphery.s.sol

```diff
@@ -0,0 +1,115 @@
+    ///         are used.
+    function name() public pure override returns (string memory) {
+        return "DeployPeriphery";
+    }
+
+    function setUp() public override {
+        super.setUp();
+
+        string memory path = string.concat(vm.projectRoot(), "/deploy-config-periphery/", deploymentContext, ".json");
+        cfg = new DeployPeripheryConfig(path);
+
+        console.log("Deploying from %s", deployScript);
+        console.log("Deployment context: %s", deploymentContext);
+    }
+
+    /// @notice Deploy all of the periphery contracts
+    function run() public {
+        console.log("Deploying all periphery contracts");
+
+        deployProxies();
+        deployImplementations();
+
+        initializeFaucet();
+    }
+
+    /// @notice Deploy all of the proxies
+    function deployProxies() public {
+        deployProxyAdmin();
+
+        deployFaucetProxy();
+    }
+
+    /// @notice Deploy all of the implementations
+    function deployImplementations() public {
+        deployFaucet();
+    }
+
+    /// @notice Modifier that wraps a function in broadcasting.
+    modifier broadcast() {
+        vm.startBroadcast();
+        _;
+        vm.stopBroadcast();
+    }
+
+    /// @notice Deploy the ProxyAdmin
+    function deployProxyAdmin() public broadcast returns (address addr_) {
+        ProxyAdmin admin = new ProxyAdmin{ salt: keccak256(bytes("proxy admin salt")) }({
+            _owner: msg.sender
+        });
+        require(admin.owner() == msg.sender);
+
+        save("ProxyAdmin", address(admin));
+        console.log("ProxyAdmin deployed at %s", address(admin));
+        addr_ = address(admin);
+    }
+
+    /// @notice Deploy the FaucetProxy
+    function deployFaucetProxy() public broadcast returns (address addr_) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy{ salt: FAUCET_IMPL_SALT }({
+            _admin: proxyAdmin
+        });
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("FaucetProxy", address(proxy));
+        console.log("FaucetProxy deployed at %s", address(proxy));
+
+        addr_ = address(proxy);
+    }
+
+    /// @notice Deploy the faucet contract.
+    function deployFaucet() public broadcast returns (address) {
+        Faucet faucet = new Faucet{ salt: FAUCET_IMPL_SALT }(cfg.faucetAdmin());
+
+        save("Faucet", address(faucet));
+        console.log("Faucet deployed at %s", address(faucet));
+
+        return address(faucet);
+    }
+
+    /// @notice Initialize the Faucet
+    function initializeFaucet() public broadcast {
+        ProxyAdmin proxyAdmin = ProxyAdmin(mustGetAddress("ProxyAdmin"));
+        address faucetProxy = mustGetAddress("FaucetProxy");
+        address faucet = mustGetAddress("Faucet");
+
+        proxyAdmin.upgrade({ _proxy: payable(faucetProxy), _implementation: faucet });
```

Comment:
Would be nice to have a require on the faucet admin here

Reply:
done